php/libraries/FilesUploadHandler.class.inc

```diff
@@ -0,0 +1,129 @@
+<?php
+/**
+ * An interface containing methods for classes used to handle file
+ * uploads from a client to a server.
+ *
+ * PHP Version 7
+ *
+ * @category Main
+ * @package  Main
+ * @author   Nicolas Brossard <nicolasbrossard.mni@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+
+/**
+ * Handles file uploads from a client to a server. The methods listed in this
+ * interface were created in order to encapsulate the usual things that happen
+ * when a set of files is uploaded to a server, namely:
+ *
+ * 1. Validation of all uploaded files
+ * 2. Transfer of all files to a designated (permanent) location on the
+ *    server.
+ * 3. Actual processing of the uploaded files (e.g. update of database
+ *    tables, etc...)
+ *
+ * @category Main
+ * @package  Main
+ * @author   Nicolas Brossard <nicolasbrossard.mni@gmail.com>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ */
+interface FilesUploadHandler
+{
+    /**
+     * Returns the properties associated to each file uploaded on the
+     * server in an array identical to $_FILES.
+     *
+     * @return array the properties for each file uploaded (one array for
+     *         each uploaded file, see $_FILES).
+     */
+    public function getFiles() : array;
+
+    /**
+     * Validates the set of uploaded files. If a given file is invalid, the
+     * validation message associated to it will be stored in the array returned
+     * by getErrors().
+     *
+     * @return bool true if the set of files is valid and should be processed, false
+     *         otherwise.
+     */
+    public function isValid() : bool;
+
+    /**
+     * Returns any validation or processing errors associated to each uploaded file.
+     *
+     * @return array list of error messages for each uploaded file.
+     */
+    public function getErrors() : array;
```

Comment:
What happens if process or validate is called multiple times? Does one error list get lost? And how do you know if it's a processing or validation error?  At the very least, this should be `getLastErrors()`, but to make things more atomic maybe the validation should return an error list rather than a bool and the process should return a (potentially different) error list rather than null.

Also, this should specify if it's talking about a numeric or associative array of error messages (and if the latter, what the key is)